cachi2/core/package_managers/gomod.py

```diff
@@ -721,35 +724,28 @@ def _get_golang_pseudo_version(
     return f"v{pseudo_semantic_version}{version_seperator}0.{commit_timestamp}-{commit_hash}"
 
 
-def _vet_local_deps(dependencies: List[dict]):
-    """Fail if any local dependency path is absolute or outside repository."""
-    for dep in dependencies:
-        version = dep["version"]
+def _validate_local_replacements(modules: List[GoModule], app_path: RootedPath) -> None:
+    replaced_modules = [
+        module for module in modules
+        if module.replace and module.replace.path.startswith(".")
+    ]
 
-        if not version:
-            continue  # go stdlib
+    for module in replaced_modules:
+        if not Path(module.replace.dir).is_relative_to(app_path.root):
+            raise Exception(f"{module.replace.dir} is not relative to {app_path.path}")
 
-        if version.startswith(".") and ".." in Path(version).parts:
-            raise UnsupportedFeature(f"Path to gomod dependency contains '..': {version}.")
-        elif version.startswith("/") or PureWindowsPath(version).root:
-            # This will disallow paths starting with '/', '\' or '<drive letter>:\'
-            raise UnsupportedFeature(
-                f"Absolute paths to gomod dependencies are not supported: {version}"
-            )
+        # Hack to set the replacement version
+        # Doing it properly will require a bigger refactoring
+        module.replace.path = _get_golang_version(module.replace.path, app_path)
 
 
-def _set_full_local_dep_relpaths(pkg_deps: List[dict], main_module_deps: List[dict]):
+def _set_locally_replaced_package_versions(pkg_deps: List[dict], main_module_deps: List[dict]):
     """
-    Set full relative paths for all local go-package dependencies.
-
-    The path that you see in the go list -deps output points only to the module that contains
-    the package. To get the full path to the package, take the relative path from the module
-    to the package (based on the package name relative to the module name) and join it with the
-    module path.
+    Set version to a package contained in a locally replaced module.
     """
-    locally_replaced_mod_names = [
-        module["name"] for module in main_module_deps if module["version"].startswith(".")
-    ]
+    modules = {
```

Comment:
Since we changed the version for the modules already (`./local-path` -> `v0.0.0`) ,  we will need to traverse all of them. Probably can be improved with some refactoring.